### PR TITLE
add github workflow and remove makefiles

### PR DESCRIPTION
**Pull request by jarlah**

### CODE REVIEW

File: pack.toml [hunk 1]
```diff
@@ -2,9 +2,16 @@
 type = "local"
 path = "."
 ipkg = "http.ipkg"
-test = "test/test.ipkg"
+test = "tests/http-test.ipkg"
+
+[custom.all.tls]
```
Comment (jarlah) [inline pack.toml]: its all kind of messy, but we need the tls so file in tests exec folder ... is this the correct way? why do we need this?

File: tests/src/ClientTest.idr [hunk 2]
```diff
@@ -87,6 +87,7 @@ test_close_without_read : EitherT String IO ()
 test_close_without_read = map_error show $ with_client {e=()} new_client_default $ \client => do
   putStrLn "sending request"
   (response, content) <- request client GET (url' "http://openbsd.org/70.html") [] ()
+  --_ <- toList_ content -- required to pass tests
```
Comment (jarlah) [inline tests/src/ClientTest.idr:90]: if we uncomment this, ~maybe~ the tests actually run ...

File: pack.toml (hunk 1 above)
Comment (jarlah) [reply pack.toml]: I thought the fix in the tls package would solve this issue. Propably not

**Comment by ReilySiegel** [comment]
Unfortunately, the tests are still using raw idris2 instead of pack, which is why you are having tls dependencies issue. This should be moved to using pack, and we can probably get rid of the makefile completely, since this package doesnt have any C support files to compile.

**Comment by jarlah** [comment]
you do see that I removed makefiles and added github workflows that uses pack right?

File: pack.toml (hunk 1 above)
Comment (jarlah) [reply pack.toml]: maybe i should try to clear pack cache or something, and try build again

**Comment by jarlah** [comment]
of course git.sr.ht had to be down now :(

**Comment by jarlah** [comment]
after removing the tls stuff from pack toml the tests still runs after removing build folder and deleting .cache folder in home folder. So this PR is good, it showcases that one of the tests never completes ... 

but the files folder for tests need to be moved, since pack run http-test doesnt find them
EDIT: moved files to root

**Comment by jarlah** [comment]
now tests passes in the fork @ReilySiegel  see https://github.com/jarlah/idris2-http/actions/runs/11640904135

but will still need to fix the failing/hanging test ofc. EDIT: not my fault originally, will need to do some bisecting

File: tests/src/ClientTest.idr (hunk 2 above)
Comment (jarlah) [reply tests/src/ClientTest.idr:90]: commented out the test in Test.idr

File: tests/src/Test.idr [hunk 3]
```diff
@@ -30,10 +30,11 @@ main = run
...
   , run_test "decompress jabberwocky.jpg.gz" test_gzip_jpg
   , run_test "decompress concatenated.gz" test_gzip_concated
-  , run_test "http close w/out read" test_close_without_read
+  --, run_test "http close w/out read" test_close_without_read
   , run_test "http cookie jar" test_cookie
   , run_test "http httpbin deflate" test_json_deflate
   , run_test "http httpbin gzip" test_json_gzip
   , run_test "http httpbin post" test_post
   , run_test "http openbsd redirect" test_redirect
+  --, run_test "chunked transfer encoding" test_chunked_transfer_encoding
```
Comment (jarlah) [inline tests/src/Test.idr:39]: bugger chunked response encoding doesnt work / is not supported

Comment (jarlah) [reply tests/src/Test.idr:39]: which perfectly explains why i couldnt make calls to discord api from by idris2 bot (small pet project)

File: .github/workflows/ci-lib.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+jobs:
+  ubuntu-chez:
+    runs-on: ubuntu-latest
+    env:
+      PACK_DIR: /root/.pack
+    strategy:
+      fail-fast: false
+    container: ghcr.io/stefan-hoeck/idris2-pack:latest
+    steps:
+      - name: Checkout idris2-tls
```
Comment (ReilySiegel) [inline .github/workflows/ci-lib.yaml]: Nit: Should be Checkout idris2-http

**Comment by ReilySiegel** [comment]
This looks ready to merge if we keep the failing tests commented out, sorry for the delay in review. Since you seem to be the primary user of this library, do you want to merge this now and fix the failing tests later, or hold off until we have resolved the underlying issues? I'm open to either approach.

**Comment by jarlah** [comment]
> This looks ready to merge if we keep the failing tests commented out, sorry for the delay in review. Since you seem to be the primary user of this library, do you want to merge this now and fix the failing tests later, or hold off until we have resolved the underlying issues? I'm open to either approach.

@ReilySiegel you can just merge it. i rebased and force pushed. Better to have a working CI setup. I can investigate the failing tests. Its a bit hard, but ill try ;)

**Pull request merged**